File: solutions/0/0/0/1/p0001_s1.c
/* Solution to Euler Problem 1: Multiples of 3 or 5. */
#include "runner.h"

/* Multiples of 3 or 5 below the limit by scanning every term and testing
 * divisibility. O(limit). */
const char *solve(int argc, char *argv[]) {
    long long max_limit = parse_int(argv[1]);
    long long result = 0;
    for (long long term = 0; term < max_limit; term++) {
        if (term % 3 == 0 || term % 5 == 0) {
            result += term;
        }
    }
    static char answer[32];
    snprintf(answer, sizeof answer, "%lld", result);
    return answer;
}

File: solutions/public/p0001/p0001_s1.py
#!/usr/bin/env python3.14
# -*- coding: utf-8 -*-
""" Solution to Euler Problem 1: Multiples of 3 or 5 [Level 0]. """
from __future__ import annotations

from solver.runners import runner


@runner.main
def solve(*args: str) -> str:
    """Multiples of 3 or 5 below the limit by scanning every term and testing
    divisibility. O(limit)."""
    max_limit: int = runner.parse_int(args[0])
    result: int = 0
    for term in range(0, max_limit):
        if term % 3 == 0 or term % 5 == 0:
            result += term
    return str(result)


if __name__ == "__main__":
    raise SystemExit(solve())
